selfdrive/ui/replay/framereader.h

```diff
@@ -5,49 +5,61 @@
 #include <atomic>
 #include <condition_variable>
 #include <mutex>
-#include <string>
 #include <thread>
 #include <vector>
 
+#include <QThread>
+
+#include "cereal/visionipc/visionbuf.h"
+
 // independent of QT, needs ffmpeg
 extern "C" {
 #include <libavcodec/avcodec.h>
 #include <libavformat/avformat.h>
 #include <libswscale/swscale.h>
 }
 
+class FrameReader : public QThread {
+  Q_OBJECT
 
-class FrameReader {
-public:
-  FrameReader(const std::string &fn);
+ public:
+  FrameReader(const std::string &url, VisionStreamType stream_type);
```

Comment:
what's the point of the vision stream type arg? it's unused

Reply:
it will be used to create vipc server buffers in https://github.com/commaai/openpilot/pull/21004

Reply:
this shouldn’t be done in framereader, see python unclogged and framereader for an example